Update mobster module

elena-buscaroli opened the pull request:
<!--
# nf-core/modules pull request

Many thanks for contributing to nf-core/modules!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the master branch.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

Closes #8792 <!-- If this PR fixes an issue, please link it here! -->

I would like to update the `mobster` module to add a feature to the output and to improve the fit of the tool.

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

SPPearce commented on modules/nf-core/mobster/templates/main_script.R:
```diff
@@ -41,14 +41,13 @@ if ( grepl(".rds\$", tolower("$rds_join")) ) {
+        input_table = lapply(names(original), function(sample_name) {
+            purity = original[[sample_name]][["purity"]]
+            original[[sample_name]] %>%
+                # keep only mutations on the diploid karyotypeCNAqc::subset_by_segment_karyotype("1:1") %>%
```
Have you lost a line break here?

SPPearce commented on modules/nf-core/mobster/templates/main_script.R:
```diff
@@ -58,11 +57,17 @@ if ( grepl(".rds\$", tolower("$rds_join")) ) {
-        dplyr::filter(VAF < 1) %>%
-        dplyr::filter(VAF!=0) %>%
-        dplyr::filter(karyotype=="1:1")
+        dplyr::rowwise() %>%
```
Why do you need rowwise here?

SPPearce commented on modules/nf-core/mobster/templates/main_script.R:
```diff
@@ -58,11 +57,17 @@ if ( grepl(".rds\$", tolower("$rds_join")) ) {
 
 ## Function to run a single mobster fit
 run_mobster_fit = function(joint_table, descr) {
+    purity = unique(joint_table[["purity"]]) %>% as.numeric()
```
Why are you doing this? What happens if there are more than one?

SPPearce replied:
Up to you, if it is never going to be a problem then you don't need one, but it doesn't hurt to add a line that throws a clear error when something goes wrong

SPPearce commented on modules/nf-core/mobster/templates/main_script.R:
```diff
@@ -58,11 +57,17 @@ if ( grepl(".rds\$", tolower("$rds_join")) ) {
-        dplyr::filter(VAF!=0) %>%
-        dplyr::filter(karyotype=="1:1")
+        dplyr::rowwise() %>%
+        dplyr::mutate(adj_VAF=VAF/purity) %>%
```
This will be the column purity not the global variable

SPPearce commented on modules/nf-core/mobster/templates/main_script.R:
```diff
@@ -58,11 +57,17 @@ if ( grepl(".rds\$", tolower("$rds_join")) ) {
+        dplyr::mutate(adj_VAF=VAF/purity) %>%
+        dplyr::ungroup() %>%
+        dplyr::filter(adj_VAF<1) %>%
+        dplyr::filter(adj_VAF>=0.05) %>%
```
Should this be a parameter?

elena-buscaroli replied:
We decided to add it as a parameter, with 0.05 as default value

SPPearce commented on modules/nf-core/mobster/templates/main_script.R:
```diff
@@ -58,11 +57,17 @@ if ( grepl(".rds\$", tolower("$rds_join")) ) {
+        dplyr::rowwise() %>%
+        dplyr::mutate(adj_VAF=VAF/purity) %>%
+        dplyr::ungroup() %>%
+        dplyr::filter(adj_VAF<1) %>%
```
Why are you filtering these?

elena-buscaroli replied:
I've removed this line which was not necessary here, thanks for noticing

SPPearce commented on modules/nf-core/mobster/templates/main_script.R:
```diff
@@ -38,31 +38,34 @@ samples = strsplit(x="$meta.tumour_sample", ",") %>% unlist()  # list of samples
+                           function(sample_name) {
+                             purity = original[[sample_name]][["purity"]]
+                             original[[sample_name]] %>%
+                               # keep only mutations on the diploid karyotypeCNAqc::subset_by_segment_karyotype("1:1") %>%
```
```suggestion
                               # keep only mutations on the diploid karyotype
                               CNAqc::subset_by_segment_karyotype("1:1") %>%
```

elena-buscaroli commented:
Thank you for the comments, I should have addressed all the comments. Also, I've updated the output names as you @SPPearce have suggested for a similar module I have released in the past. I think it is clearer in this way, but let me know.

SPPearce reviewed (approved):
Looks good to me now :)
